src/sacn/mem.c

```diff
@@ -979,28 +1043,43 @@ void deinit_sources_lost_entry(SourcesLostNotification* sources_lost)
     free(sources_lost->lost_sources);
 }
 
-void deinit_source_pcp_lost_buf(void)
+//CHRISTIAN TODO TEST!
+void deinit_sources_found_bufs(void)
 {
-  if (mem_bufs.source_pcp_lost)
-    free(mem_bufs.source_pcp_lost);
+  if (mem_bufs.sources_found)
+  {
+    for (unsigned int i = 0; i < mem_bufs.num_threads; ++i)
+      deinit_sources_found_buf(&mem_bufs.sources_found[i]);
+    free(mem_bufs.sources_found);
+  }
 }
 
-void deinit_sampling_ended_bufs(void)
+void deinit_sources_found_buf(SourcesFoundNotificationBuf* sources_found_buf)
 {
-  if (mem_bufs.sampling_ended)
+  SACN_ASSERT(sources_found_buf);
+
+  if (sources_found_buf->buf)
   {
-    for (unsigned int i = 0; i < mem_bufs.num_threads; ++i)
-      deinit_sampling_ended_buf(&mem_bufs.sampling_ended[i]);
-    free(mem_bufs.sampling_ended);
+    for (size_t i = 0; i < sources_found_buf->buf_capacity; ++i)
+    {
+      deinit_sources_found_entry(&sources_found_buf->buf[i]);
+    }
+    free(sources_found_buf->buf);
   }
 }
 
-void deinit_sampling_ended_buf(SamplingEndedNotificationBuf* sampling_ended_buf)
+void deinit_sources_found_entry(SourcesFoundNotification* sources_found)
 {
-  SACN_ASSERT(sampling_ended_buf);
+  SACN_ASSERT(sources_found);
 
-  if (sampling_ended_buf->buf)
-    free(sampling_ended_buf->buf);
+  if (sources_found->found_sources)
+    free(sources_found->found_sources);
+}
+
+void deinit_source_pap_lost_buf(void)
+{
+  if (mem_bufs.source_pap_lost)
+    free(mem_bufs.source_pap_lost);
```

Comment:
There are big sections in this file depending on #if SACN_DYNAMIC_MEM.  So I think we're covered.  But this can certainly be something you clean up as part of your work.